### ISSUE TITLE
[ComposedModal] keyboard trap fails when modal is last component in page

### ISSUE THREAD

**Issue by ptaunton** [carbon-design-system/carbon#3532]
## What package(s) are you using?
- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description
In normal use, a ComposedModal will trap the keyboard to rotate the input focus within the modal itself. If you add a ComposedModal as the last component in a page i.e. there are no naturally focusable elements in the DOM after the modal, then the focus trap fails to work.

The reason this fails to work is that the focus moves to the browser URL bar which is a non-DOM element so the `evt.relatedTarget` value in the focus event is set to `null` for security reasons. The `handleBlur` function will not work if `evt.relatedTarget` is null, so the keyboard trap fails.

This occurs in both Firefox and Chrome.

## Steps to reproduce the issue

1. Build a page with some focusable elements
2. Add a ComposedModal at the end of the page
3. Show the modal and press Tab to move the focus.
4. Observe that the focus is not trapped and actually follows through the browser URL bar and then the rest of the page.
5. Move the modal to the start of the page and observe that the focus trap now works.

**Comment by asudoh** [comment]
Hi @ptaunton thank you for reporting! You are right on what you found, and it's intentional. You can add something like `<input type="text" class="bx--assistive-text">` as the last element in `<body>`. Hope this helps!

**Comment by ptaunton** [comment]
I find it odd that this is intentional. Adding a check for a null value for `evt.relatedTarget` should be all that is needed to fix this issue. A null value will always indicate that the newly focussed item is outside the scope of the modal so it is safe to call the `focusModal` method in this case.

**Comment by asudoh** [comment]
@ptaunton We can set focus back to modal in this case, but do you think we can get focus back to our viewport in your case? e.g.:

```html
<!DOCTYPE html>
<html>
  <head>
  </head>
  <body>
    <input type="text">    
  </body>
  <script type="text/javascript">
    document.body.addEventListener('blur', (evt) => {
      if (!evt.relatedTarget) {
        document.body.querySelector('input').focus();
        console.log('Focused back to `<input>`!');
      }
    }, true)
  </script>
</html>
```

**Comment by ptaunton** [comment]
For the example above, you would need a keydown event handler and would have to stop the propagation of the TAB so that the blur never occurs since the blur is only generated after the keyboard event is processed.

**Comment by asudoh** [comment]
You are right, and the focus trap element is to avoid the extra cost of implementing the keyboard handlers.

**Comment by ptaunton** [comment]
Needing to add extra focussable elements before or after the modal is a non-obvious restriction. It also has some accessibility issues - the extra input fields would be read out by a screen reader and when there is no modal showing on the screen a keyboard user would now have a hidden focus sink on the screen.

The Angular version of carbon components doesn't have this issue. They use a keyboard-based focus cycle that works out the first and last focussable elements in the container and cycles between them. https://github.com/IBM/carbon-components-angular/blob/master/src/common/tab.service.ts

**Comment by asudoh** [comment]
I see your points, consideration is in place for better documentation, etc as well as better announcement. We are aware that there are two choices on focus-wrap, and took a route with less implementation overhead.
